rust/src/editor.rs

```diff
@@ -368,12 +376,39 @@ impl Editor {
         }
     }
 
-    pub fn do_rpc(&mut self, method: &str, params: &Value) -> Option<Value> {
+    fn delete_to_end_of_paragraph(&mut self, kill_ring: &Mutex<Rope>) {
+        let current = self.view.sel_max();
+        let offset = self.cursor_end_offset();
+        let mut val = String::from("");
+
+        if current != offset {
+            val = self.text.slice_to_string(current, offset);
```

Comment:
This feels like unneeded duplicate code. But I'll merge this because overall it looks quite good, and this can be a cleanup for later. Thanks!